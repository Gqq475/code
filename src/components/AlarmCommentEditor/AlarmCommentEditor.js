import React from 'react'
import classes from './AlarmCommentEditor.scss'
import { connect } from 'react-redux'
import { compose, withState } from 'recompose'
import OurToaster from 'components/OurToaster'

const qca = __QCA__

type Props = {
  id: Number,
  content: String,
  token: String,
  btnIsDisabled: Boolean,
  onSubmitNewComment: Function
};

export class AlarmCommentEditor extends React.Component {
  props: Props;

  constructor (props) {
    super(props)
    this.onChangeComment = this.onChangeComment.bind(this)
    this.handleChange = this.handleChange.bind(this)
    this.state = {
      text: this.props.content
    }
  }

  onChangeComment () {
    let newComment = document.getElementById('commentTA').value
    // this.props.disableBtn(true)

    fetch(`${qca}/alarm/updateAlarm`, {
      method: 'PUT',
      body: JSON.stringify({
        id: this.props.id,
        comments: newComment
      }),
      headers: {
        'Authorization': this.props.token
      }
    })
    .then(response => response.json())
    .then(json => {
      if (json.code === 200000) {
        // 成功
        this.props.onSubmitNewComment(newComment)
      } else {
        OurToaster.show({ message: '意见更新失败!' })
        // this.props.disableBtn(false)
      }
    })
    .catch(() => {
      OurToaster.show({ message: '意见更新失败!' })
      // this.props.disableBtn(false)
    })
  }

  componentWillUnmount () {

  }

  handleChange (e) {
    this.setState({ text: e.target.value })
  }

  render () {
    return (
      <div className={classes['AlarmCommentEditor-container']}>
        <label className={classes.textLabel}>意见</label>
        <textarea id='commentTA' dir='auto'
          className={`pt-input ${classes.textarea}`}
          value={this.state.text} onChange={this.handleChange} />
        <div className={classes.button}>
          <button
            type='button'
            disabled={this.props.btnIsDisabled}
            onClick={this.onChangeComment}
            className='pt-button pt-minimal pt-icon-confirm' />
        </div>
      </div>
    )
  }
}

export default compose(
  connect((state) => ({
    token: state.auth.token
  })),
  withState('btnIsDisabled', 'disableBtn', false)
)(AlarmCommentEditor)
