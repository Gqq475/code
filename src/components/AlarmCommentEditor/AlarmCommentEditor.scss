.AlarmCommentEditor-container {
  display: flex;
  flex-direction: column;

  .button {
    margin: 0 0 0 auto;
  }

  .textLabel {
    color: #909090;
    width: 40%;
    padding: 5px 20px 5px 0;
    text-align: right;
  }

  .textarea {
    margin-left: 20px;
  }
}
