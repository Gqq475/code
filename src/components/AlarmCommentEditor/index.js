import AlarmCommentEditor from './AlarmCommentEditor'
export default AlarmCommentEditor
