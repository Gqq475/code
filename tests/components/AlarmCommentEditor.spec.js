import React from 'react'
import { AlarmCommentEditor } from 'components/AlarmCommentEditor/AlarmCommentEditor'

describe('(Component) AlarmCommentEditor', () => {
  it('render normally', () => {
    const wrapper = shallow(
      <AlarmCommentEditor />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
