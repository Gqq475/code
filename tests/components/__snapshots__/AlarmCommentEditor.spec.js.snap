// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) AlarmCommentEditor render normally 1`] = `
<div
  className="AlarmCommentEditor-container"
>
  <label
    className="textLabel"
  >
    意见
  </label>
  <textarea
    className="pt-input textarea"
    dir="auto"
    id="commentTA"
    onChange={[Function]}
  />
  <div
    className="button"
  >
    <button
      className="pt-button pt-minimal pt-icon-confirm"
      onClick={[Function]}
      type="button"
    />
  </div>
</div>
`;
